### PR TITLE
Add cache store option to config

### PR DESCRIPTION
**Pull request by tlaverdure**
- [ ] Added or updated tests
- [x] Documented user facing changes
- [X] Updated CHANGELOG.md

Need to use a different cache store for AST due to cache item size limitations.

Somewhat related to https://github.com/nuwave/lighthouse/issues/1389

**Changes**

Adds an option to use a different cache store from the default. 

**Breaking changes**

N/A

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ You can find and compare releases at the [GitHub release page](https://github.co
 
 - Add `@withCount` directive to eager load relationship counts on field access https://github.com/nuwave/lighthouse/pull/1390
 - Extend `lighthouse:directive` artisan command to allow choosing interfaces https://github.com/nuwave/lighthouse/pull/1251
+- Add `lighthouse.cache.store` configuration option to set the cache store to use for shema caching https://github.com/nuwave/lighthouse/pull/1446
```
Comment (spawnia) [inline CHANGELOG.md:16]:
```suggestion
- Add `lighthouse.cache.store` configuration option to set the cache store to use for schema caching https://github.com/nuwave/lighthouse/pull/1446
```

Comment (tlaverdure) [reply CHANGELOG.md:16]: 😅

File: src/lighthouse.php [hunk 2]
```diff
@@ -85,6 +85,7 @@
     'cache' => [
         'enable' => env('LIGHTHOUSE_CACHE_ENABLE', env('APP_ENV') !== 'local'),
         'key' => env('LIGHTHOUSE_CACHE_KEY', 'lighthouse-schema'),
+        'store' => env('LIGHTHOUSE_CACHE_STORE', null),
```
Comment (spawnia) [inline src/lighthouse.php:88]: Can you add a small docblock to each option?

This one might be:

```suggestion

        /*
         * Allows using a specific cache store, uses the app's default if set to null.
         */
        'store' => env('LIGHTHOUSE_CACHE_STORE', null),
```

**Comment by spawnia** [comment]
We might include a hint that file-based caching works well when deploying immutable container images, maybe in the docs? Our app currently has no need for other caches, but if we were to add them, i would still keep up that separation.

File: src/lighthouse.php [hunk 3]
```diff
@@ -83,8 +83,25 @@
...
+         */
         'key' => env('LIGHTHOUSE_CACHE_KEY', 'lighthouse-schema'),
+
+        /**
+         * Allows using a specific cache store, uses the app's default if set to null.
+         */
+        'store' => env('LIGHTHOUSE_CACHE_STORE', null),
+
+        /**
+         * Indicates how long the schema should remain cached. A null value
```
Comment (spawnia) [inline src/lighthouse.php:102]:
```suggestion
         * Duration in seconds the schema should remain cached, null means forever.
```

File: src/lighthouse.php [hunk 4]
```diff
@@ -83,8 +83,25 @@
     */
 
     'cache' => [
+        /**
```
Comment (spawnia) [inline src/lighthouse.php:86]:
```suggestion
        /*
```

Because they are not actual PHPDocs.

File: src/lighthouse.php [hunk 5]
```diff
@@ -83,8 +83,25 @@
...
         'key' => env('LIGHTHOUSE_CACHE_KEY', 'lighthouse-schema'),
+
+        /*
+         * Allows using a specific cache store, uses the app's default if set to null.
+         */
+        'store' => env('LIGHTHOUSE_CACHE_STORE', null),
+
+        /**
+         * Duration in seconds the schema should remain cached, null means forever.
+         * ensures the schema is cached forever or until cleared.
```
Comment (spawnia) [inline src/lighthouse.php:103]:
```suggestion
```

**Comment by tlaverdure** [comment]
Yeah adding that hint in the docs would be good.

**Pull request merged**